rasa/utils/tensorflow/models.py

```diff
@@ -95,55 +80,13 @@ def __init__(
 
         self._training = None  # training phase should be defined when building a graph
 
-        self._predict_function = None
-
         self.random_seed = random_seed
 
-        self.tensorboard_log_dir = tensorboard_log_dir
-        self.tensorboard_log_level = tensorboard_log_level
-
-        self.train_summary_writer = None
-        self.test_summary_writer = None
-        self.model_summary_file = None
-        self.tensorboard_log_on_epochs = True
-
-        self.best_metrics_so_far = {}
-        self.checkpoint_model = checkpoint_model
-        self.best_model_file = None
-        self.best_model_epoch = -1
-        if self.checkpoint_model:
-            model_checkpoint_dir = rasa.utils.io.create_temporary_directory()
-            self.best_model_file = os.path.join(
-                model_checkpoint_dir, f"{CHECKPOINT_MODEL_NAME}.tf_model"
-            )
-
-    def _set_up_tensorboard_writer(self) -> None:
-        if self.tensorboard_log_dir is not None:
-            if self.tensorboard_log_level not in TENSORBOARD_LOG_LEVELS:
-                raise ValueError(
-                    f"Provided '{TENSORBOARD_LOG_LEVEL}' "
-                    f"('{self.tensorboard_log_level}') "
-                    f"is invalid! Valid values are: {TENSORBOARD_LOG_LEVELS}"
-                )
-            self.tensorboard_log_on_epochs = self.tensorboard_log_level == "epoch"
-
-            current_time = datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
-            class_name = self.__class__.__name__
-
-            train_log_dir = (
-                f"{self.tensorboard_log_dir}/{class_name}/{current_time}/train"
-            )
-            test_log_dir = (
-                f"{self.tensorboard_log_dir}/{class_name}/{current_time}/test"
-            )
+        random.seed(self.random_seed)
+        tf.random.set_seed(self.random_seed)
+        np.random.seed(self.random_seed)
```

Comment:
I'd say it's better to move this three function calls out of constructor to a separate function.